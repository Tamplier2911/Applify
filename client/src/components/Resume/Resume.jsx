// import "./Resume.scss";
import React from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";

// JS Rendering CSS
import {
  ResumeContainer,
  ResumeHeader,
  ResumeHeaderImgContainer,
  ResumeHeaderImg,
  ResumeHeaderTitle,
  ResumeHeaderName,
  ResumeHeaderOccupation,
  ResumeInfo,
  ResumeInfoAbout,
  ResumeInfoDetails,
  ResumeInfoWrapper,
  ResumeInfoContacts,
  ResumeInfoContacstWrapper,
  ResumeInfoContacsLink,
  ResumeMain,
  ResumeMainExperience,
  ResumeMainEducation,
  ResumeMainEducationOrg,
  ResumeMainEducationOrgDate,
  ResumeMainEducationOrgName,
  ResumeSkills,
  ResumeMainSkills,
  ResumeMainSkillsProg,
  ResumeMainSkillsProgWrapper,
  ResumeMainInterests,
  ResumeMainInterestsCover,
  ResumeMainInterestsWrapper,
  ResumeLinksContainer,
  ResumeDownloadLink,
  ResumeBlackHeader,
  ResumeBlueHeader,
  ResumeSubTitle,
  ResumeMailSVG,
  ResumePhoneSVG,
  ResumeChromeSVG,
  ResumeTwitterSVG,
  ResumeCorrectSVG,
  ResumeProgressContainer,
  ResumeProgressCircle,
  ResumeProgressText,
} from "./ResumeStyles";

// resume for download
import resEng from "../../assets/pdf/Artem_Nikolaiev_CV_ENG.pdf";
import resRus from "../../assets/pdf/Artem_Nikolaiev_CV_RUS.pdf";
import resUkr from "../../assets/pdf/Artem_Nikolaiev_CV_UKR.pdf";

// component constants
import resumeData from "./ResumeConstants";

const Resume = ({ lang }) => {
  const {
    header: { name, occupation, image },
    info: {
      about: { aboutHeader, aboutText },
      details: {
        detailsHeader,
        birthTitle,
        birthDate,
        nationalityTitle,
        nationality,
        addressTitle,
        address,
      },
      contacts: { contactsHeader, email, phone, github, twitter },
    },
    main: {
      experience: { experienceHeader, experienceTitle, experienceText },
      education: { educationHeader, educationCollection },
      skills: {
        skillsHeaderFront,
        skillListFront,
        skillsHeaderBack,
        skillListBack,
      },
      interests: { interestsHeader, interestSubjects },
    },
    downloadLinks: { linkDescriptions },
  } = resumeData[lang];

  return (
    <ResumeContainer lang={lang}>
      <ResumeHeader animateIn={"bounceInUp"} animateOnce={true}>
        <ResumeHeaderImgContainer>
          <ResumeHeaderImg alt="author" src={image} />
        </ResumeHeaderImgContainer>
        <ResumeHeaderTitle>
          <ResumeHeaderName>{name}</ResumeHeaderName>
          <ResumeHeaderOccupation>{occupation}</ResumeHeaderOccupation>
        </ResumeHeaderTitle>
      </ResumeHeader>
      <ResumeInfo animateIn={"bounceInRight"} animateOnce={true}>
        <ResumeInfoAbout>
          <ResumeBlackHeader>{aboutHeader}</ResumeBlackHeader>
          <ResumeSubTitle>{aboutText}</ResumeSubTitle>
        </ResumeInfoAbout>
        <ResumeInfoDetails>
          <ResumeBlackHeader>{detailsHeader}</ResumeBlackHeader>
          <ResumeInfoWrapper>
            <ResumeBlueHeader>{birthTitle}</ResumeBlueHeader>
            <ResumeSubTitle>{birthDate}</ResumeSubTitle>
          </ResumeInfoWrapper>
          <ResumeInfoWrapper>
            <ResumeBlueHeader>{nationalityTitle}</ResumeBlueHeader>
            <ResumeSubTitle>{nationality}</ResumeSubTitle>
          </ResumeInfoWrapper>
          <ResumeInfoWrapper>
            <ResumeBlueHeader>{addressTitle}</ResumeBlueHeader>
            <ResumeSubTitle>{address}</ResumeSubTitle>
          </ResumeInfoWrapper>
        </ResumeInfoDetails>
        <ResumeInfoContacts>
          <ResumeBlackHeader>{contactsHeader}</ResumeBlackHeader>
          <ResumeInfoContacstWrapper>
            <ResumeMailSVG />
            <ResumeSubTitle>{email}</ResumeSubTitle>
          </ResumeInfoContacstWrapper>
          <ResumeInfoContacstWrapper>
            <ResumePhoneSVG />
            <ResumeSubTitle>{phone}</ResumeSubTitle>
          </ResumeInfoContacstWrapper>
          <ResumeInfoContacstWrapper>
            <ResumeChromeSVG />
            <ResumeInfoContacsLink
              href={github}
              rel="noopener noreferrer"
              target="_blank"
            >
              GitHub
            </ResumeInfoContacsLink>
          </ResumeInfoContacstWrapper>

          <ResumeInfoContacstWrapper>
            <ResumeTwitterSVG />
            <ResumeInfoContacsLink
              href={twitter}
              rel="noopener noreferrer"
              target="_blank"
            >
              Twitter
            </ResumeInfoContacsLink>
          </ResumeInfoContacstWrapper>
        </ResumeInfoContacts>
      </ResumeInfo>
      <ResumeMain animateIn={"bounceInLeft"} animateOnce={true}>
        <ResumeMainExperience>
          <ResumeBlackHeader>{experienceHeader}</ResumeBlackHeader>
          <ResumeBlueHeader>{experienceTitle}</ResumeBlueHeader>
          <ResumeSubTitle>{experienceText}</ResumeSubTitle>
        </ResumeMainExperience>
        <ResumeMainEducation>
          <ResumeBlackHeader>{educationHeader}</ResumeBlackHeader>
          <ResumeMainEducationOrg>
            <ResumeBlueHeader>
              {educationCollection[0].educationSubject}
            </ResumeBlueHeader>
            <ResumeMainEducationOrgDate>
              {educationCollection[0].educationDate}
            </ResumeMainEducationOrgDate>
            <ResumeMainEducationOrgName>
              {educationCollection[0].educationOrg}
            </ResumeMainEducationOrgName>
          </ResumeMainEducationOrg>
          <ResumeMainEducationOrg>
            <ResumeBlueHeader>
              {educationCollection[1].educationSubject}
            </ResumeBlueHeader>
            <ResumeMainEducationOrgDate>
              {educationCollection[1].educationDate}
            </ResumeMainEducationOrgDate>
            <ResumeMainEducationOrgName>
              {educationCollection[1].educationOrg}
            </ResumeMainEducationOrgName>
          </ResumeMainEducationOrg>
        </ResumeMainEducation>
        <ResumeSkills>
          <ResumeMainSkills>
            <ResumeBlackHeader>{skillsHeaderFront}</ResumeBlackHeader>
            <ResumeMainSkillsProg>
              <ResumeMainSkillsProgWrapper>
                <ResumeProgressContainer
                  percentage={skillListFront[0].skillPercentage}
                >
                  <ResumeProgressCircle>
                    <ResumeProgressText>
                      {skillListFront[0].skillPercentage}%
                    </ResumeProgressText>
                  </ResumeProgressCircle>
                </ResumeProgressContainer>
                <ResumeSubTitle>
                  {skillListFront[0].skillSubject}
                </ResumeSubTitle>
              </ResumeMainSkillsProgWrapper>
              <ResumeMainSkillsProgWrapper>
                <ResumeProgressContainer
                  percentage={skillListFront[1].skillPercentage}
                >
                  <ResumeProgressCircle>
                    <ResumeProgressText>
                      {skillListFront[1].skillPercentage}%
                    </ResumeProgressText>
                  </ResumeProgressCircle>
                </ResumeProgressContainer>
                <ResumeSubTitle>
                  {skillListFront[1].skillSubject}
                </ResumeSubTitle>
              </ResumeMainSkillsProgWrapper>
            </ResumeMainSkillsProg>
          </ResumeMainSkills>
          <ResumeMainSkills>
            <ResumeBlackHeader>{skillsHeaderBack}</ResumeBlackHeader>
            <ResumeMainSkillsProg>
              <ResumeMainSkillsProgWrapper>
                <ResumeProgressContainer
                  percentage={skillListBack[0].skillPercentage}
                >
                  <ResumeProgressCircle>
                    <ResumeProgressText>
                      {skillListBack[0].skillPercentage}%
                    </ResumeProgressText>
                  </ResumeProgressCircle>
                </ResumeProgressContainer>
                <ResumeSubTitle>{skillListBack[0].skillSubject}</ResumeSubTitle>
              </ResumeMainSkillsProgWrapper>
              <ResumeMainSkillsProgWrapper>
                <ResumeProgressContainer
                  percentage={skillListBack[1].skillPercentage}
                >
                  <ResumeProgressCircle>
                    <ResumeProgressText>
                      {skillListBack[1].skillPercentage}%
                    </ResumeProgressText>
                  </ResumeProgressCircle>
                </ResumeProgressContainer>
                <ResumeSubTitle>{skillListBack[1].skillSubject}</ResumeSubTitle>
              </ResumeMainSkillsProgWrapper>
            </ResumeMainSkillsProg>
          </ResumeMainSkills>
        </ResumeSkills>
        <ResumeMainInterests>
          <ResumeBlackHeader>{interestsHeader}</ResumeBlackHeader>
          <ResumeMainInterestsCover>
            <ResumeMainInterestsWrapper>
              <ResumeCorrectSVG />
              <ResumeSubTitle>
                {interestSubjects[0].interestDescription}
              </ResumeSubTitle>
            </ResumeMainInterestsWrapper>
            <ResumeMainInterestsWrapper>
              <ResumeCorrectSVG />
              <ResumeSubTitle>
                {interestSubjects[1].interestDescription}
              </ResumeSubTitle>
            </ResumeMainInterestsWrapper>
            <ResumeMainInterestsWrapper>
              <ResumeCorrectSVG />
              <ResumeSubTitle>
                {interestSubjects[2].interestDescription}
              </ResumeSubTitle>
            </ResumeMainInterestsWrapper>
            <ResumeMainInterestsWrapper>
              <ResumeCorrectSVG />
              <ResumeSubTitle>
                {interestSubjects[3].interestDescription}
              </ResumeSubTitle>
            </ResumeMainInterestsWrapper>
          </ResumeMainInterestsCover>
        </ResumeMainInterests>
      </ResumeMain>
      <ResumeLinksContainer animateIn={"bounceInDown"} animateOnce={true}>
        <ResumeDownloadLink href={resEng} download>
          {linkDescriptions[0].text}
        </ResumeDownloadLink>
        <ResumeDownloadLink href={resRus} download>
          {linkDescriptions[1].text}
        </ResumeDownloadLink>
        <ResumeDownloadLink href={resUkr} download>
          {linkDescriptions[2].text}
        </ResumeDownloadLink>
      </ResumeLinksContainer>
    </ResumeContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage,
});

export default connect(mapStateToProps)(Resume);
