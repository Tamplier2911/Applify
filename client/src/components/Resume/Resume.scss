/*
.resume-page {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  // justify-self: center;
  padding: 0 2rem;
  grid-row-gap: 4rem;
  grid-column-gap: 4rem;

  // grid-row-gap: 15rem;
  // grid-template-rows: 70vh 1fr;
  // grid-row-gap: 10rem;

  // @media only screen and (max-width: 980px) {
  //   grid-template-rows: 68vh 1fr;
  //   grid-row-gap: 0rem;
  // }

  // @media only screen and (max-width: 768px) {
  //   grid-template-rows: 65vh 1fr;
  // }

  // @media only screen and (max-width: 720px) {
  //   grid-template-rows: 60vh 1fr;
  //   grid-row-gap: 7rem;
  // }

  @media only screen and (max-width: 768px) {
    grid-row-gap: 8rem;
  }

  @media only screen and (max-width: 540px) {
    grid-row-gap: 4rem;
  }

  @media only screen and (max-width: 425px) {
    grid-row-gap: 2rem;
  }

  // header
  &__header {
    grid-column: 1 / -1;
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    // grid-column-gap: 15rem;
    grid-column-gap: 4rem;
    grid-row-gap: 2rem;
    align-items: center;
    justify-items: center;

    &--image-container {
      grid-column: 1 / 2;
      height: 15rem;
      width: 15rem;
      border-radius: 50%;
      overflow: hidden;
      display: flex;

      @media only screen and (max-width: 540px) {
        height: 12rem;
        width: 12rem;
      }

      @media only screen and (max-width: 310px) {
        grid-column: 1 / -1;
      }
    }

    &--image {
      width: 100%;
      height: 100%;
      display: block;
      object-fit: cover;
    }

    &--title {
      grid-column: 2 / -1;

      @media only screen and (max-width: 310px) {
        grid-column: 1 / -1;
      }
    }

    &--name {
      color: var(--cl-font-resume);
      font-size: 6rem;
      line-height: 0.9;

      @media only screen and (max-width: 720px) {
        font-size: 4rem;
      }

      @media only screen and (max-width: 310px) {
        font-size: 3rem;
      }
    }

    &--occupation {
      font-size: 1.7rem;
      text-transform: uppercase;

      @media only screen and (max-width: 720px) {
        font-size: 1.5rem;
      }

      @media only screen and (max-width: 310px) {
        font-size: 1.4rem;
        text-align: center;
      }
    }
  }

  // info
  &__info {
    grid-column: 1 / 2;
    display: grid;
    grid-auto-rows: min-content;
    grid-row-gap: 2rem;

    @media only screen and (max-width: 720px) {
      grid-column: 1 / -1;
    }

    &--about {
      &-title {
        font-family: var(--font-title);
        text-transform: uppercase;
        font-weight: bold;
      }

      &-text {
        color: var(--cl-darkest-grey);
      }
    }

    &--details {
      line-height: 1.5;

      @media only screen and (max-width: 720px) {
        display: grid;
      }

      &-wrap {
        @media only screen and (max-width: 720px) {
          display: grid;
          grid-column-gap: 2rem;
          grid-template-columns: 1fr 1fr;
        }

        @media only screen and (max-width: 540px) {
          grid-column-gap: 0rem;
          grid-template-columns: 1fr;
        }
      }
    }

    &--contact {
      display: grid;
      grid-row-gap: 0.4rem;

      @media only screen and (max-width: 720px) {
        grid-template-columns: 1fr 1fr;
        grid-column-gap: 1rem;

        & h4 {
          grid-column: 1 / -1;
        }
      }

      @media only screen and (max-width: 540px) {
        grid-template-columns: 1fr;
        grid-column-gap: 0rem;
      }

      &-wrapper {
        display: grid;
        align-items: center;
        grid-template-columns: min-content 1fr;
        grid-column-gap: 1rem;
      }

      &-icon {
        background-color: var(--cl-font-resume);
        border-radius: 50%;
        height: 2.5rem;
        width: 2.5rem;
      }

      &-link {
        &:link,
        &:visited {
          font-family: var(--font-title);
          color: var(--cl-darkest-grey);
          text-decoration: none;
          transition: color 0.3s;
        }

        &:hover,
        &:active {
          color: var(--cl-font-resume);
        }
      }
    }
  }

  // main
  &__main {
    grid-column: 2 / -1;
    display: grid;
    grid-row-gap: 3rem;

    @media only screen and (max-width: 720px) {
      grid-column: 1 / -1;
    }

    &--education {
      &-org {
        display: grid;
        grid-template-columns: 1fr min-content;
        &:not(:first-child) {
          margin-top: 1rem;
        }

        @media only screen and (max-width: 360px) {
          grid-template-columns: 1fr;
          // grid-row-gap: 0.5rem;
        }

        &-name {
          text-transform: uppercase;
          font-weight: bold;
          font-size: 1.4rem;

          @media only screen and (max-width: 360px) {
            order: 2;
          }
        }
        &-date {
          // grid-row: 1 / -1;
          white-space: nowrap;
          font-family: var(--font-title);
          color: var(--cl-darkest-grey);
          @media only screen and (max-width: 360px) {
            order: 3;
          }
        }
      }
    }

    &--skills {
      display: grid;
      grid-template-columns: 1fr 1fr;

      @media only screen and (max-width: 540px) {
        grid-template-columns: 1fr;
        grid-row-gap: 2rem;
      }

      &-front-progress {
        display: grid;
        grid-row-gap: 1rem;
      }

      &-front-progress-wrap {
        display: grid;
        grid-template-columns: min-content 1fr;
        align-items: center;
        grid-column-gap: 1rem;
      }

      &-front-circleprogress {
        width: 4rem;
        height: 4rem;
        background-color: var(--cl-font-resume);
        border-radius: 50%;
      }

      &-back-circleprogress {
        width: 4rem;
        height: 4rem;
        background-color: var(--cl-font-resume);
        border-radius: 50%;
      }
    }

    &--interesets {
      &-cover {
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-row-gap: 0.5rem;

        @media only screen and (max-width: 540px) {
          grid-template-columns: 1fr;
          grid-row-gap: 0.5rem;
        }
      }

      &-wrapper {
        display: grid;
        grid-template-columns: min-content 1fr;
        grid-column-gap: 1rem;
      }

      &-icon {
        width: 2.5rem;
        height: 2.5rem;
        background-color: var(--cl-font-resume);
        border-radius: 50%;
      }
      &-intereset {
      }
    }
  }
}

.links-container {
  grid-column: 1 / -1;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-column-gap: 2rem;
  // justify-items: center;
  margin: 4rem 0;

  @media only screen and (max-width: 620px) {
    grid-template-columns: 1fr;
    grid-column-gap: 0rem;
    grid-row-gap: 2rem;
  }

  &__link {
    &:link,
    &:visited {
      font-weight: bold;
      text-decoration: none;
      color: var(--cl-font-resume);
      border: 0.2rem solid var(--cl-font-resume);
      padding: 0.5rem 2rem;
      text-align: center;
      transition: color 0.3s, background-color 0.3s, border 0.3s;
    }

    &:hover,
    &:active {
      color: var(--cl-white);
      background-color: var(--cl-font-resume);
    }
  }
}

// 360 deg / 100 pices * percent vlaue = required value
.progress {
  &__container {
    display: flex;
    align-items: center;
    justify-content: center;

    width: 5rem;
    height: 5rem;
    border-radius: 50%;
    background-image: linear-gradient(0deg, #ddd 50%, transparent 50%),
      linear-gradient(-50deg, #028cd5 50%, #ddd 50%);
  }

  &__circle {
    display: flex;
    align-items: center;
    justify-content: center;

    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    background-color: #fff;
  }

  &__middle {
    font-weight: bold;
    color: var(--cl-font-resume);
    font-size: 1.2rem;
  }
}

*/
