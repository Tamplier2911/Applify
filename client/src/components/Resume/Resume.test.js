import React from "react";
import toJson from "enzyme-to-json";

// component
import Resume from "./Resume";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<Resume />", () => {
  const initialState = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialState);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<Resume store={store} />);
  });

  it("renders without an error", () => {
    const resume = findByTestAttr(wrapper, "resume");
    expect(resume).toHaveLength(1);
  });

  it("renders children - resume head", () => {
    const resumeHead = findByTestAttr(wrapper, "resume-head");
    expect(resumeHead).toHaveLength(1);
  });

  it("renders children - resume info", () => {
    const resumeInfo = findByTestAttr(wrapper, "resume-info");
    expect(resumeInfo).toHaveLength(1);
  });

  it("renders children - resume main", () => {
    const resumeMain = findByTestAttr(wrapper, "resume-main");
    expect(resumeMain).toHaveLength(1);
  });

  it("renders children - resume links", () => {
    const resumeLinks = findByTestAttr(wrapper, "resume-links");
    expect(resumeLinks).toHaveLength(1);
  });

  it("have expected href and text content - github link ", () => {
    const resumeGithubLink = findByTestAttr(wrapper, "resume-github");
    expect(resumeGithubLink).toHaveLength(1);
    expect(resumeGithubLink.text()).toBe("GitHub");
    expect(resumeGithubLink.prop("href")).toEqual(
      "https://github.com/Tamplier2911"
    );
  });

  it("have expected href and text content - twitter link", () => {
    const resumeTwitterLink = findByTestAttr(wrapper, "resume-twitter");
    expect(resumeTwitterLink).toHaveLength(1);
    expect(resumeTwitterLink.text()).toBe("Twitter");
    expect(resumeTwitterLink.prop("href")).toEqual(
      "https://twitter.com/ArtyomNikolayev"
    );
  });

  it("renders children - resumne ", () => {
    const resume = findByTestAttr(wrapper, "resume");
    expect(toJson(resume)).toMatchSnapshot();
  });
});
