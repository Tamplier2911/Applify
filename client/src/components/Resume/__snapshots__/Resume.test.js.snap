// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Resume /> renders children - resumne  1`] = `
<styled.div
  data-test="resume"
  lang="eng"
>
  <Styled(ScrollAnimation)
    animateIn="bounceInUp"
    animateOnce={true}
    data-test="resume-head"
  >
    <styled.div>
      <styled.img
        alt="author"
        src="profile-portrait.jpg"
      />
    </styled.div>
    <styled.div>
      <styled.h2>
        Artem Nikolaiev
      </styled.h2>
      <styled.h4>
        Front-End Engineer
      </styled.h4>
    </styled.div>
  </Styled(ScrollAnimation)>
  <Styled(ScrollAnimation)
    animateIn="bounceInRight"
    animateOnce={true}
    data-test="resume-info"
  >
    <styled.div>
      <styled.h4>
        About me
      </styled.h4>
      <styled.div>
        My passion - technologies and self-education, looking forward along-term productive partnership.
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.h4>
        Personal Details
      </styled.h4>
      <styled.div>
        <styled.h4>
          Birth Date
        </styled.h4>
        <styled.div>
          15 / 03 / 1990
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.h4>
          Nationality
        </styled.h4>
        <styled.div>
          Ukrainian
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.h4>
          Address
        </styled.h4>
        <styled.div>
          Avdiivka, Donetska Oblast, Ukraine
        </styled.div>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.h4>
        Contacts
      </styled.h4>
      <styled.div>
        <Styled(Component) />
        <styled.div>
          artyom.nikolaev@yahoo.com
        </styled.div>
      </styled.div>
      <styled.div>
        <Styled(Component) />
        <styled.div>
          +38 093 151 21 28
        </styled.div>
      </styled.div>
      <styled.div>
        <Styled(Component) />
        <styled.a
          data-test="resume-github"
          href="https://github.com/Tamplier2911"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </styled.a>
      </styled.div>
      <styled.div>
        <Styled(Component) />
        <styled.a
          data-test="resume-twitter"
          href="https://twitter.com/ArtyomNikolayev"
          rel="noopener noreferrer"
          target="_blank"
        >
          Twitter
        </styled.a>
      </styled.div>
    </styled.div>
  </Styled(ScrollAnimation)>
  <Styled(ScrollAnimation)
    animateIn="bounceInLeft"
    animateOnce={true}
    data-test="resume-main"
  >
    <styled.div>
      <styled.h4>
        Professional Experience
      </styled.h4>
      <styled.h4>
        Front-end Engineer
      </styled.h4>
      <styled.div>
        My experience in web development mainly comes from self-education and independant projects, that I love to challenge myself with. Having 12+ month of experience studying and building web solutions, I have developed a strong passion for this domain. To demonstrate my learning curve, below I have provided a git link, which you can use to review the projects I have built along the way. This is my first application for Front-end dev. position, and I am excited to put my skill to the test as a part of growing, ambitious team.
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.h4>
        Education
      </styled.h4>
      <styled.div>
        <styled.h4>
          Computer Science 50
        </styled.h4>
        <styled.div>
          2018 - 2019
        </styled.div>
        <styled.div>
          edx.org - online courses
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.h4>
          Bachelor of Restaurant and Hotel Business
        </styled.h4>
        <styled.div>
          2009 - 2014
        </styled.div>
        <styled.div>
          Donetsk National University of Economics and Trade
        </styled.div>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.h4>
          Front-end Skills
        </styled.h4>
        <styled.div>
          <styled.div>
            <styled.div
              percentage={76}
            >
              <styled.div>
                <styled.div>
                  76
                  %
                </styled.div>
              </styled.div>
            </styled.div>
            <styled.div>
              HTML / CSS / JS
            </styled.div>
          </styled.div>
          <styled.div>
            <styled.div
              percentage={70}
            >
              <styled.div>
                <styled.div>
                  70
                  %
                </styled.div>
              </styled.div>
            </styled.div>
            <styled.div>
              React.js
            </styled.div>
          </styled.div>
          <styled.div>
            <styled.div
              percentage={35}
            >
              <styled.div>
                <styled.div>
                  35
                  %
                </styled.div>
              </styled.div>
            </styled.div>
            <styled.div>
              React Native
            </styled.div>
          </styled.div>
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.h4>
          Back-end Skills
        </styled.h4>
        <styled.div>
          <styled.div>
            <styled.div
              percentage={50}
            >
              <styled.div>
                <styled.div>
                  50
                  %
                </styled.div>
              </styled.div>
            </styled.div>
            <styled.div>
              Node.js / Express
            </styled.div>
          </styled.div>
          <styled.div>
            <styled.div
              percentage={39}
            >
              <styled.div>
                <styled.div>
                  39
                  %
                </styled.div>
              </styled.div>
            </styled.div>
            <styled.div>
              MongoDB / Mongoose
            </styled.div>
          </styled.div>
          <styled.div>
            <styled.div
              percentage={20}
            >
              <styled.div>
                <styled.div>
                  20
                  %
                </styled.div>
              </styled.div>
            </styled.div>
            <styled.div>
              Docker / Kubernetes
            </styled.div>
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.h4>
        Interests
      </styled.h4>
      <styled.div>
        <styled.div>
          <Styled(Component) />
          <styled.div>
            Music & Movies
          </styled.div>
        </styled.div>
        <styled.div>
          <Styled(Component) />
          <styled.div>
            Technologies
          </styled.div>
        </styled.div>
        <styled.div>
          <Styled(Component) />
          <styled.div>
            Reading & Education
          </styled.div>
        </styled.div>
        <styled.div>
          <Styled(Component) />
          <styled.div>
            Sitting with cocoa covered in blanket
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </Styled(ScrollAnimation)>
  <Styled(ScrollAnimation)
    animateIn="bounceInDown"
    animateOnce={true}
    data-test="resume-links"
  >
    <styled.a
      download={true}
      href="Artem_Nikolaiev_CV_ENG.pdf"
    >
      Download in English
    </styled.a>
    <styled.a
      download={true}
      href="Artem_Nikolaiev_CV_RUS.pdf"
    >
      Download in Russian
    </styled.a>
    <styled.a
      download={true}
      href="Artem_Nikolaiev_CV_UKR.pdf"
    >
      Download in Ukrainian
    </styled.a>
  </Styled(ScrollAnimation)>
</styled.div>
`;
